### PR TITLE
User has the ability to login.

### PR DESCRIPTION
**Pull request by JaimeCrz**
User story we are working:
Status: (Delivered)

on:
https://www.pivotaltracker.com/story/show/171561429

```
As a restaurant API
In order to welcome previous users to continue their order
I would like to give them the ability to login
```
We have added a User class capable of authenticate an account thanks to the newly installed gem Devise Token Auth. The class has been tested and passed the following test:

- Sign in with valid credentials.
- Sign in with invalid password.
- Sign in with invalid email.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+
```
Comment (faraznaeem) [inline app/models/user.rb]: please mind the formatting

File: config/initializers/devise_token_auth.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+
+
```
Comment (faraznaeem) [inline config/initializers/devise_token_auth.rb]: formatting

File: config/routes.rb [hunk 3]
```diff
@@ -1,7 +1,14 @@
 Rails.application.routes.draw do
+  mount_devise_token_auth_for 'User', at: 'api/v1/auth', skip: [:omniauth_callbacks]
   namespace :api do
+
     namespace :v0 do
```
Comment (faraznaeem) [inline config/routes.rb]: are these namespaces necessary? and do you really want a pings controller here?

File: spec/support/response_json.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+module  ResponseJSON
+    def response_json
+        JSON.parse(response.body)
+
+    end
```
Comment (faraznaeem) [inline spec/support/response_json.rb:4]: formatting

**Review by faraznaeem — CHANGES_REQUESTED** [review]

File: config/routes.rb (hunk 3 above)
Comment (JaimeCrz) [reply config/routes.rb]: We decided to remove the pings controllers line + test + controller.

**Review by oliverochman — APPROVED** [review]

**Pull request merged**